core/src/main/scala/zquery/ZQuery.scala

```diff
@@ -271,7 +287,7 @@ object ZQuery {
    * Constructs a query from an effect.
    */
   final def fromEffect[R, E, A](effect: ZIO[R, E, A]): ZQuery[R, E, A] =
-    ZQuery(effect.map(Result.done))
+    ZQuery(effect.fold(e => Result.fail(Cause.fail(e)), a => Result.done(a)))
```

Comment:
Wondering if we should use `foldCause` to wrap the `die` case?

Reply:
Yes.